src/main/sensors/battery.c

```diff
@@ -248,6 +254,112 @@ static void batteryUpdateVoltageState(void)
 
 }
 
+////////
+#ifdef USE_VBAT_ALARM_DELAY
+			//			https ://github.com/betaflight/betaflight/issues/1887
+			//			https ://github.com/betaflight/betaflight/pull/6691
+// uses unfilterd 
+static void batteryUpdateVoltageStateDelay(timeUs_t currentTimeUs)
+{
+	// alerts are currently used by beeper, osd and other subsystems
+	switch (voltageState) {
+	case BATTERY_OK:
```

Comment:
Copying the logic to update the battery voltage state is not a good approach - this will make maintaining this code hard, as all changes will have to be made in two places. You should much rather add support for the delay to the existing logic.

Reply:
the issue 
 batteryUpdateVoltageState(timeUs_t currentTimeUs ) would have a compile warning of unused  currentTimeUs

Reply:
All good, there is no hurry - let us know what you find out.